### DISCUSSION TITLE
Some ROCM 5.3.0 Packages Require Clean Chroot

### DISCUSSION THREAD

**Discussion by Mushoz** [rocm-arch/rocm-arch#898] [General]
I have been using the ROCM stack for as long as I can remember, and always used yay to update without any issues whatsoever. That changed with the latest 5.3.0 release. I now HAVE to build in a clean chroot, which was not previously required (even though it has always been recommended). Using the suggested way that is described in the Arch Wiki is quite frankly quite painful and not a good user experience at all.

I ended up trying to build/install via Paru, but I am not happy with that solution for two reasons:
1. I am now forced to use Paru, even though Yay has been very nice to me all this time.
2. It's unrelated to these ROCM packages, but following the Paru instructions I was STILL unable to build in a clean chroot.

At the end of the day, I ended up using the environment variables that were described in this Repo's README. Now on to my question:
1. Would it not be possible to add these environment variable to the PKGBUILD? If the rocm-llvm dependency is known to conflict with non-rocm llvm, it makes sense to explicitly define the correct one, right?
2. Are there any downsides / potential regressions to this suggestion? It should obviously still cleanly build in a clean chroot, but if that is still the case, then I don't see any downsides to adding these environment variables to the PKGBUILD. Thoughts?

**Comment by Maetveis** [comment]
For the `hsa-rocr` and clang-14 problem I get why @tpkessler doesn't want to take on maintenance of the LLVM and Clang cmake directories.
In the arch spirit of fixing stuff upstream I thought about adding version checks to [ROCR-Runtime][hsa-rocr] in a PR.
However that turns out to be hard because clang does not provide `ClangConfigVersion.cmake`, and even if it did llvm only promises compatibility with the same major version, so it would be a maintenance burden for upstream to take such a patch.

That basically leaves no choice for upstream (barring considerably larger effort), except to hope for whatever clang version is selected it's capable of compiling for the new target architectures. 

Upstream has one more escape hatch that could be used: `CMAKE_PREFIX_PATH`, if set to `/opt/rocm` the rocm-arch clang and llvm would be found before the system one. I think this  wouldn't be a high maintenance burden to take on.

Reiterating @Mushoz points, was https://github.com/rocm-arch/rocm-arch/pull/885 rejected only _because_ it provides a fix for an unsupported use-case (building without chroot) or because of another reason?

If the first, could this decision be reconsidered? As evident from the issues and discussions here many users expect these packages to compile even without a chroot so it could be supported on a best effort basis.  Even if the maintainers of this repository don't agree with it and therefore don't test such a configuration, it would receive testing and bug fixes from the community.

[hsa-rocr]: https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/adae6c61e10d371f7cbc3d0e94ae2c070cab18a4/src/core/runtime/trap_handler/CMakeLists.txt#L46

**Reply by tpkessler** [reply]
Hey @Maetveis! Thank you very much for the analysis of the cmake files in `hsa-rocr`. As you already said, I didn't accept the PR as it adds additional complexity to the PKGBUILD for an issue that only appears in a specific (albeit common) user configuration. Top priority for me are fixes for issues that break packages built in a clean chroot. But I really like your suggestion to add `/opt/rocm` to `CMAKE_PREFIX_PATH`. It's a meaningful addition to the `cmake` call, fixes the specific problem and may avoid further problems in the future. The new version in the AUR should also build in user space without a clean chroot.

**Comment by tpkessler** [comment]
Following the suggestion by @Maetveis I've added `/opt/rocm` to the cmake prefix path. This fixes the specific problem without being complicated to maintain.
